CRM/Dedupe/Merger.php

```diff
@@ -730,9 +733,14 @@ public static function merge($dupePairs = array(), $cacheParams = array(), $mode
     // doNotResetCache flag
     $config = CRM_Core_Config::singleton();
     $config->doNotResetCache = 1;
+    $deletedContacts = array();
 
     while (!empty($dupePairs)) {
-      foreach ($dupePairs as $dupes) {
+      foreach ($dupePairs as $index => $dupes) {
+        if (in_array($dupes['dstID'], $deletedContacts) || in_array($dupes['srcID'], $deletedContacts)) {
+          unset($dupePairs[$index]);
```

Comment:
I think it would be useful to add something into the message that's displayed to say that x contacts were not merged, because their duplicate had already been deleted.

ie:
4 contacts merged
2 pairs skipped due to contact deletion

Although that's not very descriptive. Or is this something for a follow-up?